pkg/apis/serving/v1beta1/revision_defaults.go

```diff
@@ -44,48 +44,46 @@ func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
 		rs.TimeoutSeconds = &ts
 	}
 
-	var container corev1.Container
-	if len(rs.PodSpec.Containers) == 1 {
-		container = rs.PodSpec.Containers[0]
+	if len(rs.PodSpec.Containers) == 0 {
+		rs.PodSpec.Containers = make([]corev1.Container, 1)
```

Comment:
I am feeling that we do not need to set defaults when no container is specified, because it fails eventually due to missing image. But I added this as current code does this.

Reply:
I think I agree with you here. Just because we used to do it doesn't mean we should keep doing it. This seems unnecessary and I would be okay removing it.